src/lib.ts

```diff
@@ -1112,6 +1116,88 @@ function head(list: NixType): NixType {
   return listStrict.values[0];
 }
 
+function all(pred: NixType): NixType {
+  const lambdaStrict = pred.toStrict();
+  if (!(lambdaStrict instanceof Lambda)) {
+    throw new EvalException(
+      `'all' function requires another function, but got '${lambdaStrict.typeOf()}' instead.`,
+    );
+  }
+
+  return new Lambda((list) => {
+    const listStrict = list.toStrict();
+    if (!(listStrict instanceof NixList)) {
+      throw new EvalException(
+        `Cannot apply the 'all' function on '${listStrict.typeOf()}'.`,
+      );
+    }
+
+    for (const element of listStrict.values) {
+      const result = lambdaStrict.apply(element);
+      if (!result.asBoolean()) {
+        return FALSE;
+      }
+    }
+
+    return TRUE;
+  });
+}
+
+function any(pred: NixType): NixType {
+  const lambdaStrict = pred.toStrict();
+  if (!(lambdaStrict instanceof Lambda)) {
+    throw new EvalException(
+      `'any' function requires another function, but got '${lambdaStrict.typeOf()}' instead.`,
+    );
+  }
+
+  return new Lambda((list) => {
+    const listStrict = list.toStrict();
+    if (!(listStrict instanceof NixList)) {
+      throw new EvalException(
+        `Cannot apply the 'any' function on '${listStrict.typeOf()}'.`,
+      );
+    }
+
+    for (const element of listStrict.values) {
+      const result = lambdaStrict.apply(element);
+      if (result.asBoolean()) {
+        return TRUE;
+      }
+    }
+
+    return FALSE;
+  });
+}
+
+function attrNames(set: NixType): NixType {
+  const listStrict = set.toStrict();
+  if (!(listStrict instanceof Attrset)) {
+    throw new EvalException(
+      `Cannot apply the 'attrNames' function on '${listStrict.typeOf()}'.`,
+    );
+  }
+
+  const keys = Array.from(listStrict.keys());
+  keys.sort();
+
+  return new NixList(keys.map((key) => new NixString(key)));
+}
+
+function attrValues(set: NixType): NixType {
+  const listStrict = set.toStrict();
+  if (!(listStrict instanceof Attrset)) {
+    throw new EvalException(
+      `Cannot apply the 'attrValues' function on '${listStrict.typeOf()}'.`,
+    );
+  }
+
+  const keys = Array.from(listStrict.keys());
+  keys.sort();
```

Comment:
Oh, interesting, nice catch about otdering. Just checking if this is how Nix actually does it. Does it always return keys in lexicographic order? Would you have any links to documentation (or code) that shows this?

Reply:
https://nixos.org/manual/nix/stable/language/builtins.html

Mentions they're alphabetically sorted